keras/layers/attention/attention.py

```diff
@@ -174,8 +174,8 @@ def _apply_scores(self, scores, value, scores_mask=None, training=False):
             weights = backend.random.dropout(
                 weights,
                 self.dropout,
-                noise_shape=self.noise_shape,
-                seed=self.seed_generator,
+                noise_shape=None,
+                seed=None,
```

Comment:
But `self.seed_generator` not initialized anywhere in Attention layer and will raise `'Attention' object has no attribute 'seed_generator'`. Either we can set it to None as proposed above or we need to initialize it in `AttentionLayer` or `Layer` API.

Either we can do this in `Attention` layer like below: 

```
class Attention(Layer):
  def __init__(self,*,**kwargs):
      self.noise_shape = kwargs.pop('noise_shape',None)
      self.seed_generator = kwargs.pop('seed_generator',None)
      super().__init__(**kwargs)
      #remaining code follows
```
Here we need to initialize this before `super()` call else it will raise error from here.
https://github.com/keras-team/keras/blob/233761df6184e87e1ec5eb512e5c87534a33d141/keras/layers/layer.py#L262-L266

Or in Layer API like below:
```
class Layer(BackendLayer, Operation):
    def __init__(self,*,**kwargs):
      self.noise_shape = kwargs.pop('noise_shape',None)
      self.seed_generator = kwargs.pop('seed_generator',None)
      #remaining code follows
```

If we want to pass noise_shape or seed_generator to Attention layer we need to initialize them explicitly which seems missing now.